hkmc2/shared/src/main/scala/hkmc2/codegen/js/JSBuilder.scala

```diff
@@ -417,3 +422,18 @@ object JSBuilder:
 end JSBuilder
 
 
+trait JSBuilderSanityChecks(instrument: Bool) extends JSBuilder:
+  
+  val functionParamVarargSymbol = semantics.TempSymbol(0, N, "args")
+  
+  override def setupFunction(params: List[semantics.Param], body: Block)(using Raise, Scope): (Document, Document) =
+    if instrument then
+      val paramsList = params.map(p => Scope.scope.allocateName(p.sym))
+      val paramsStr = Scope.scope.allocateName(functionParamVarargSymbol)
+      val checkArgsNum = doc"if ($paramsStr.length !== ${params.length}) { throw new globalThis.Error('got ' + $paramsStr.length + ' arguments, expecting ' + ${params.length}) }\n"
```

Comment:
I would change this boilerplate to:
```suggestion
      val checkArgsNum = doc"globalThis.Predef.checkArgs($functionName, $paramsStr.length, ${params.length})"
```
with the appropriate helper method defined in `Predef.mls`.

Reply:
Thanks! I have pushed the commit to improve the error message. As for the helper method `checkArgs` in `Predef.mls`, there seems to be no way to throw errors in `*.mls` files..? Sorry that I vaguely remember that you told me about this in our previous meeting, but cannot recall the detail.

Reply:
Oh, do you mean that we're missing the `throw` keyword? This should be a trivial thing to add (it's already supported internally by the IR), but we can do it in a separate PR.

Reply:
In case you want to do it: I would add `throw` as a symbol in `Elaborator.unaryOps` and handle it specially in `Lowering`.

Reply:
Yes, I mean the `throw` keyword is missing, and yes I would like to do this. I just took a look, and it seems that `Elaborator.unaryOps` is only used [here](https://github.com/hkust-taco/mlscript/blob/e3859934e2c38db20f2a6f6998aba55dc49940a9/hkmc2/shared/src/main/scala/hkmc2/semantics/Elaborator.scala#L105-L107), which assumes all unaryOps must also be binary ops, so maybe there are more changes needed? Maybe I can look into how `return` is handled and do something similar?

Reply:
I tried to implement the support for `throw` following how `return` is handled and created a new PR https://github.com/hkust-taco/mlscript/pull/239

Reply:
Thanks, merged!

Reply:
Thanks! I have pushed the changes to use `Predef.checkArgs`.